test/unit/commands/recordscreen-specs.js

```diff
@@ -0,0 +1,135 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import AndroidDriver from '../../..';
+import _ from 'lodash';
+import { withMocks } from 'appium-test-support';
+import ADB from 'appium-adb';
+import { sleep } from 'asyncbox';
+
+chai.should();
+chai.use(chaiAsPromised);
+
+describe('recording the screen', () => {
+  let driver = new AndroidDriver();
+  let adb = new ADB();
+  driver.adb = adb;
+
+  describe('recording the screen', withMocks({adb, driver}, (mocks) => {
```

Comment:
This test does not do what it is supposed to do. It should be testing `driver.startRecordingScreen` but it just goes through the steps of that method and mocks everything. Nothing is being tested at all.

The tests ought to mock the calls that `startRecordingScreen` makes, to direct the flow in particular directions (a test for each possible direction), to test the logic within `startRecordingScreen` itself. As it is, that method is mocked, and so nothing inside is tested.

So, for instance, the first two error conditions are exercised by:
```js
      it('should fail to recording the screen on an emulator', async function () {
        mocks.driver.expects('isEmulator').returns(true);
        await driver.startRecordingScreen().should.eventually.be.rejectedWith(/Android screen recording does not work on emulators/);
      });
      it('should fail to recording the screen on a device with API level 18', async function () {
        mocks.driver.expects('isEmulator').returns(false);
        mocks.adb.expects('getApiLevel').returns(18);
        await driver.startRecordingScreen().should.eventually.be.rejectedWith(/Unable to record screen on API level '18'. Minimum level is 19/);
      });
```